examples/PageLocale.re

```diff
@@ -1,16 +1,6 @@
-let pageLocale =
-  [@intl.messages]
-  {
-    "hello": {
-      "id": "page.hello",
-      "defaultMessage": "Hello",
-    },
-    "world": {
-      "id": "page.world",
-      "defaultMessage": "World",
-    },
-    "today": {
-      "id": "page.today",
-      "defaultMessage": "Today is",
-    },
-  };
+open ReactIntl;
+[@intl.messages];
+
+let hello = {id: "page.hello", defaultMessage: "Hello"};
```

Comment:
Is there any way to generate ids with PPX too?

Reply:
Which PPX are you referring to? bs-react-intl-extractor is not a PPX and can only extract existing messages.

But yes, it might be nice to have a PPX that could extract a message with id `PageLocale.hello` and defaultMessage `Hello` from module `PageLocale.re` with

```reason
let hello = [%intl.msg "Hello"];
```

or something like that.

Reply:
> Which PPX are you referring to?

There is not such ppx yet, it exists only in my dreams :) But maybe I can find time to work on it. I an ppx noob, but I think it should be possible to use ppx-per file, not ppx-per-constant.
[@intl.someppx];
let hello = {id: "page.hello", defaultMessage: "Hello", description: None};
let world = "world";

Reply:
I would not specify the id explicitly, but just use `<module name>.<variable name>` as the id automatically.

Anyway, let's leave this for a possible future PR. 😉

Reply:
👌  I think I saw ppx that does this id thing.

It can even be

```reason
let a = [%intl.msg "message a"];
let b = [%intl.msg {message: "message b", description: "description b"}];
```

Reply:
@alexfedoseev this is great! But is it possible to use ppx only once at the beginning of the module?

```
[%intl.msgmodule]
let a = {message: a};
let b = {message: "message b", description: "description b"};
```

Reply:
@ixzzd I don't think it's possible since PPX wouldn't capture outside things. I'm not sure how exactly `[@bs.config {jsx: 2}]` works but guessing it's an attribute that is used by the compiler to figure how to handle the current module, i.e. it's not handled by the PPX engine. The same way as the extractor consumes `[@intl.messages]`.